ui/src/components/entity/citation/PaperAbstract.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from "react";
+import { Paper } from "../../../api/types";
+import { truncateText } from "../../../utils/ui";
+
+interface Props {
+  paper: Paper;
+}
+
+interface State {
+  showFullAbstract: boolean;
+}
+
+export default class PaperAbstract extends React.Component<Props, State> {
+  state = {
+    showFullAbstract: false,
+  };
+
+  toggleFullAbstract = () => {
+    this.setState((prev) => {
+      return {
+        showFullAbstract: !prev.showFullAbstract,
+      };
+    });
+  };
+
+  render() {
+    const { paper } = this.props;
+    const { showFullAbstract } = this.state;
+    const truncatedAbstract = paper.abstract
+      ? truncateText(paper.abstract, 300)
+      : null;
+    const abstractsMatch = truncatedAbstract === paper.abstract;
+    const expandMsg = showFullAbstract ? "(show less)" : "(show more)";
+    return (
+      <p className="paper-summary__abstract">
+        <>
+          {showFullAbstract || abstractsMatch
+            ? paper.abstract
+            : truncatedAbstract}
+          {!abstractsMatch && (
+            <span
+              className="paper-summary__abstract__show-more-label"
+              onClick={this.toggleFullAbstract}
+            >
+              {expandMsg}
+            </span>
+          )}
+        </>
+      </p>
```

Comment:
Where does the space between the abstract and the more label come from? Or does the text touch?

Reply:
They do touch...the ellipsis used to shorten the abstract does create a little more visual space vs. other characters though. I'll add a touch of padding on the label.